src/fwrite.c

```diff
@@ -566,7 +566,7 @@ int compressbuff(z_stream *stream, void* dest, size_t *destLen, const void* sour
 
   stream->next_out = dest;
   stream->avail_out = *destLen;
-  stream->next_in = (z_const Bytef *)source;
+  stream->next_in = (Bytef *)source; // don't use z_const anywhere; #3939
```

Comment:
What was wrong with `const` exactly?

I see this in my `zlib.h`:

```
/usr/include/zconf.h:234:#if defined(ZLIB_CONST) && !defined(z_const)
/usr/include/zconf.h:235:#  define z_const const
/usr/include/zconf.h-236-#else
/usr/include/zconf.h:237:#  define z_const
/usr/include/zconf.h-238-#endif
```

would it make sense to simply repeat this logic like

```
#ifndef zconst
#define z_const
#endif 
```

?

Reply:
I put the error message I got in the comment at the top of this issue.
I don't understand what z_const is for.  If we put those 3 lines back in, could you think of a comment to put alongside to explain why if z_const is defined, it should be used?  And why const can't be used?

Reply:
Sorry I missed the comment, I had only seen the commit message.

I don't really know any details, mine was more a question of curiosity... perhaps @philippechataignon could comment based on motivation for using `z_const` in the first place?